### DISCUSSION TITLE
How do I get $watch working with x-for checkbox values?

### DISCUSSION THREAD

**Discussion by zarahzachz** [alpinejs/alpine#1371] [1. Help] [answered]
This is my first time using Alpine and I'm trying to make a todo app to see how it works, but I've been struggling to get my `$watch` function to watch changes happening to checkboxes inside an `x-for` loop.

Code example: https://codepen.io/zarahzachz/pen/MWpXpXZ

I'd like my todo list data to be saved to `localStorage` so a user doesn't lose what's been done on a page refresh or something. I was able to get `localStorage` working when adding new items to the list, but it's not watching changes on the list items after that - namely it doesn't update the 'done' value to reflect the checkbox state. 

I added an `x-model` to the checkbox that appears to signal the changes correctly (seen in some `<pre>` tags I'm rendering on the page), but a page refresh doesn't show this change being persisted and I'm kinda stumped.

This code's probably entirely wrong, so any advice/feedback is 💯 appreciated!

**Comment by zarahzachz** [comment]
I think I fixed this... by adding `@change="todos = todos"` to my checkboxes. This is probably entirely wrong, but I'm getting the correct results!

**Reply by zarahzachz** [reply]
J/K thought I had the solution and this isn't working quite as I'd hoped. Selecting multiple checkboxes only persists a handful. 

While this solution mostly works, it seems like there's a delay that happens after checking each checkbox, so the state doesn't get updated consistently.

**Comment by KevinBatdorf** [comment, marked as answer]
You have the `x-model` assigned to a property that doesn't exist. `x-model="todo.done"` where `todo` is only a local variable on the object you're iterating. You might want, `x-model="todos[index].done"`

**Reply by zarahzachz** [reply]
Thanks for the reply! 

I was hopeful this would work, but I'm getting the same results as before - I can get the checked state to save if I check each checkbox slowly, but if I check everything in one go none of them save.

Attempted `x-model="todos[index]['done']` and `x-model="todos[index].done`.

**Reply by KevinBatdorf** [reply]
`$watch` doesn't watch property changes like that so you would need another solution. you can do `todos = todos` or maybe have a "lastUpdated" property that you watch.

**Reply by zarahzachz** [reply]
Sounds good! Thanks for helping me understand my issue a little more!
